2023-08-29 14:10:26,438 - worker_process - INFO - Worker_process started.
2023-08-29 14:10:26,681 - Cell - INFO - site-1.fcf43644-75f6-49a1-aac9-ab501c87485b: created backbone internal connector to tcp://localhost:1078 on parent
2023-08-29 14:10:26,690 - Cell - INFO - site-1.fcf43644-75f6-49a1-aac9-ab501c87485b: created backbone external connector to grpc://192.168.100.3:8102
2023-08-29 14:10:26,690 - nvflare.fuel.f3.sfm.conn_manager - INFO - Connector [CH00001 ACTIVE tcp://localhost:1078] is starting
2023-08-29 14:10:26,690 - nvflare.fuel.f3.sfm.conn_manager - INFO - Connector [CH00002 ACTIVE grpc://192.168.100.3:8102] is starting
2023-08-29 14:10:26,691 - FederatedClient - INFO - Wait for client_runner to be created.
2023-08-29 14:10:26,691 - FederatedClient - INFO - Got client_runner after 0.0003058910369873047 seconds
2023-08-29 14:10:26,692 - FederatedClient - INFO - Got the new primary SP: grpc://192.168.100.3:8102
2023-08-29 14:10:26,701 - ClientRunner - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b]: client runner started
2023-08-29 14:10:26,729 - Communicator - INFO - Received from secure_project server  (1114733 Bytes). getTask: train time: 0.018170833587646484 seconds
2023-08-29 14:10:26,730 - FederatedClient - INFO - pull_task completed. Task name:train Status:True 
2023-08-29 14:10:26,730 - ClientRunner - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b, peer=secure_project, peer_run=fcf43644-75f6-49a1-aac9-ab501c87485b]: got task assignment: name=train, id=05555722-beeb-4f12-ae8e-511f77ac5324
2023-08-29 14:10:26,731 - ClientRunner - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b, peer=secure_project, peer_run=fcf43644-75f6-49a1-aac9-ab501c87485b, task_name=train, task_id=05555722-beeb-4f12-ae8e-511f77ac5324]: invoking task executor LearnerExecutor
2023-08-29 14:10:26,731 - LearnerExecutor - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b, peer=secure_project, peer_run=fcf43644-75f6-49a1-aac9-ab501c87485b, task_name=train, task_id=05555722-beeb-4f12-ae8e-511f77ac5324]: Client trainer got task: train
2023-08-29 14:10:26,731 - MNISTLearner - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b, peer=secure_project, peer_run=fcf43644-75f6-49a1-aac9-ab501c87485b, task_name=train, task_id=05555722-beeb-4f12-ae8e-511f77ac5324]: Client site-1 initialized at 
 /home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/../fcf43644-75f6-49a1-aac9-ab501c87485b/app_site-1 
 with args: Namespace(workspace='/home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/..', startup='/home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/../startup', token='46f86140-8de3-4521-841f-6972de9e2a26', ssid='ebc6125d-0a56-4688-9b08-355fe9e4d61a', job_id='fcf43644-75f6-49a1-aac9-ab501c87485b', client_name='site-1', sp_target='192.168.100.3:8102', parent_url='tcp://localhost:1078', fed_client='fed_client.json', set=['secure_train=true', 'uid=site-1', 'org=nvidia', 'config_folder=config', 'print_conf=True'], local_rank=0, train_config='config/config_train.json', client_config='config/config_fed_client.json', config_folder='config', env='config/environment.json')
2023-08-29 14:10:27,993 - ClientRunner - ERROR - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b, peer=secure_project, peer_run=fcf43644-75f6-49a1-aac9-ab501c87485b, task_name=train, task_id=05555722-beeb-4f12-ae8e-511f77ac5324]: Processing error from executor LearnerExecutor: ValueError: num_samples should be a positive integer value, but got num_samples=0
2023-08-29 14:10:27,993 - ClientRunner - ERROR - Traceback (most recent call last):
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/private/fed/client/client_runner.py", line 254, in _process_task
    reply = executor.execute(task.name, task.data, fl_ctx, self.task_abort_signal)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 86, in execute
    return self.train(shareable, fl_ctx, abort_signal)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 99, in train
    validate_result: Shareable = self.learner.validate(shareable, fl_ctx, abort_signal)
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 355, in validate
    self._create_datasets(fl_ctx)
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 173, in _create_datasets
    self.train_loader = self.train_datamgr.get_loader(self.transform_train, batch_size=self.batch_size, shuffle=True, num_workers=self.num_workers)
  File "/home/aienode1/NVFlare/sound_cls/pt/utils/data_manager.py", line 34, in get_loader
    return data.DataLoader(dataset=dataset, shuffle=shuffle, batch_size=batch_size, num_workers=num_workers, drop_last=True, collate_fn=self.pad_seq)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 351, in __init__
    sampler = RandomSampler(dataset, generator=generator)  # type: ignore[arg-type]
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/sampler.py", line 107, in __init__
    raise ValueError("num_samples should be a positive integer "
ValueError: num_samples should be a positive integer value, but got num_samples=0

2023-08-29 14:10:27,995 - FederatedClient - INFO - Starting to push execute result.
2023-08-29 14:10:27,998 - Communicator - INFO -  SubmitUpdate size: 575 Bytes. time: 0.0033087730407714844 seconds
2023-08-29 14:10:27,999 - ClientRunner - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b, peer=secure_project, peer_run=fcf43644-75f6-49a1-aac9-ab501c87485b, task_name=train, task_id=05555722-beeb-4f12-ae8e-511f77ac5324]: result sent to server for task: name=train, id=05555722-beeb-4f12-ae8e-511f77ac5324
2023-08-29 14:10:28,958 - ClientRunner - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b, peer=secure_project, peer_run=fcf43644-75f6-49a1-aac9-ab501c87485b]: received aux request from Server to end current RUN
2023-08-29 14:10:28,958 - ClientRunner - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b]: ABORT (RUN) command received
2023-08-29 14:10:28,959 - ClientRunner - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b]: ABORT (RUN) requests end run events sequence
2023-08-29 14:10:28,959 - ClientRunner - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b]: ABOUT_TO_END_RUN fired
2023-08-29 14:10:28,960 - ClientRunner - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b]: END_RUN fired
2023-08-29 14:10:30,001 - ClientRunner - INFO - [identity=site-1, run=fcf43644-75f6-49a1-aac9-ab501c87485b]: run method requests end run events sequence
2023-08-29 14:10:30,016 - FederatedClient - INFO - Shutting down client run: site-1
2023-08-29 14:10:32,050 - MPM - INFO - MPM: Good Bye!
